src/Disks/ObjectStorages/DiskObjectStorageTransaction.cpp

```diff
@@ -330,8 +360,8 @@ struct RemoveRecursiveObjectStorageOperation final : public IDiskObjectStorageOp
 
                 if (unlink_outcome && !file_names_remove_metadata_only.contains(rel_path))
                 {
-                    objects_to_remove_by_path[std::move(rel_path)]
-                        = ObjectsToRemove{std::move(objects_paths), std::move(unlink_outcome)};
+                    local_paths_to_remove.push_back(std::move(rel_path));
```

Comment:
What is the purpose of splitting to the vectors ?

Reply:
Two vectors should consume less memory than map, although after reducing map size it is not that significant

Reply:
So I suggest we go back to the map